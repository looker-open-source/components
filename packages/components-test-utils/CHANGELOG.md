# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
