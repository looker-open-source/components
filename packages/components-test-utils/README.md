# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
