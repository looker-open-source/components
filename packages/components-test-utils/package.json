{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
