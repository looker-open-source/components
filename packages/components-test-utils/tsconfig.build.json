{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
