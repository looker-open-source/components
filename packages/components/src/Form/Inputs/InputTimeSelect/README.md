# InputTimeSelect

`InputTimeSelect` provides a [`Select`](/components/forms/select)-based dropdown UI for picking a 24-hour time value. This is an alternative to [`InputTime`](/components/forms/input-time), which provides an HTML5-standard time picking experience.

The default implementation provides a dropdown list of times in 15-minute increments, formatted in for a 12-hour clock.

```tsx
function Basic(props: InputTimeSelectProps) {
  return <InputTimeSelect {...props} />;
}
```

## Intervals

You can choose from any of the following options: 5, 10, 15, 20, 30, or 60 minute intervals.

We've limited allowed intervals to a few approved options rather than accepting any number. This is to protect against uninteded use where an engineer might specify 1-minute intervals and generate an unusable select box with 1400 options. We also wouldn't want to support bizarre instances of a random numbers like 37 or 152 minute intervals.

```tsx
function Intervals(props: InputTimeSelectProps) {
  const { interval: _intervals, ...restProps } = props;
  return (
    <Fieldset>
      <SpaceVertical gap="u1">
        <Heading as="h4">5-minute</Heading>
        <InputTimeSelect interval={5} {...restProps} />
      </SpaceVertical>
      <SpaceVertical gap="u1">
        <Heading as="h4">10-minute</Heading>
        <InputTimeSelect interval={10} {...restProps} />
      </SpaceVertical>
      <SpaceVertical gap="u1">
        <Heading as="h4">15-minute</Heading>
        <InputTimeSelect interval={15} {...restProps} />
      </SpaceVertical>
      <SpaceVertical gap="u1">
        <Heading as="h4">60-minute</Heading>
        <InputTimeSelect interval={20} {...restProps} />
      </SpaceVertical>
      <SpaceVertical gap="u1">
        <Heading as="h4">30-minute</Heading>
        <InputTimeSelect interval={30} {...restProps} />
      </SpaceVertical>
      <SpaceVertical gap="u1">
        <Heading as="h4">60-minute</Heading>
        <InputTimeSelect interval={60} {...restProps} />
      </SpaceVertical>
    </Fieldset>
  );
}
```

## 12- or 24-hour formatting

Depending on readability preference, the time picker can format the labels in either 12- or 24-hour time. Regardless of label formatting, this component returns a 24-hour time formatted string when changed.

```tsx
function TimeFormatting(props: InputTimeSelectProps) {
  const { format: _format, value: valueProp = '14:00', ...restProps } = props;

  const [value, setValue] = useState<string | undefined>(valueProp);
  return (
    <Grid>
      <SpaceVertical gap="u2">
        <Heading as="h3">12-hour</Heading>
        <Space>
          <InputTimeSelect
            format="12h"
            value={value}
            onChange={setValue}
            {...restProps}
          />
        </Space>
      </SpaceVertical>
      <SpaceVertical gap="u2">
        <Heading as="h3">24-hour</Heading>
        <Space>
          <InputTimeSelect
            format="24h"
            value={value}
            onChange={setValue}
            {...restProps}
          />
        </Space>
      </SpaceVertical>
    </Grid>
  );
}
```

## Controlled Component

Consistent with all other inputs, `InputTimeSelect` accepts `value` and `onChange` props for use in a controlled form setting.

- `value` accepts a 24-hour formatted time string (e.g. `15:45`)
- `onChange` returns a 24-hour time string

```tsx
function Controlled(props: InputTimeSelectProps) {
  const { value: valueProp = '14:00', ...restProps } = props;

  const [value, setValue] = useState<string | undefined>(valueProp);

  return (
    <Space>
      <InputTimeSelect value={value} onChange={setValue} {...restProps} />
      <Paragraph color="text2">{value}</Paragraph>
    </Space>
  );
}
```

## DefaultValue

If you don't want to control the form value externally, but still want to listen for changes, you can still pass in an initial `defaultValue`.

```tsx
function DefaultValue(props: InputTimeSelectProps) {
  const { defaultValue = '15:45', ...restProps } = props;
  return <InputTimeSelect defaultValue={defaultValue} {...restProps} />;
}
```
