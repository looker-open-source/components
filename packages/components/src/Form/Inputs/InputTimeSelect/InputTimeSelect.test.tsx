/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import { fireEvent } from '@testing-library/react'
import { renderWithTheme } from '@looker/components-test-utils'
import map from 'lodash/map'

import { InputTimeSelect, InputTimeSelectProps } from './InputTimeSelect'

const realDateNow = Date.now.bind(global.Date)
// eslint-disable-next-line @typescript-eslint/unbound-method
const realScrollIntoView = window.HTMLElement.prototype.scrollIntoView

beforeEach(() => {
  // eslint-disable-next-line @typescript-eslint/unbound-method
  window.HTMLElement.prototype.scrollIntoView = jest.fn()
  // eslint-disable-next-line @typescript-eslint/unbound-method
  global.Date.now = jest.fn(() => 1580538819172)
})

afterEach(() => {
  // eslint-disable-next-line @typescript-eslint/unbound-method
  global.Date.now = realDateNow // reset Date.now mock
  // eslint-disable-next-line @typescript-eslint/unbound-method
  window.HTMLElement.prototype.scrollIntoView = realScrollIntoView
  jest.resetAllMocks()
})

// parse text content from options list
const extractTextFromDomList = (list: HTMLElement) => {
  const options = list.getElementsByTagName('li')
  return map(options, (el: HTMLElement) => {
    return el.textContent
  })
}

const renderListContent = (props: InputTimeSelectProps) => {
  const { getByPlaceholderText, getByRole } = renderWithTheme(
    <InputTimeSelect {...props} />
  )

  // open list
  const inputBox = getByPlaceholderText('Select time')
  fireEvent.click(inputBox)

  // select list container
  const domList = getByRole('listbox')
  return domList
}

describe('prop: format', () => {
  test('formats options in 12 hour time', () => {
    const domList = renderListContent({ format: '12h' })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })

  test('formats options in 24 hour time', () => {
    const domList = renderListContent({ format: '24h' })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })
})

describe('prop: interval', () => {
  test('renders 5-minute intervals', () => {
    const domList = renderListContent({ interval: 5 })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })

  test('renders 10-minute intervals', () => {
    const domList = renderListContent({ interval: 10 })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })
  test('renders 15-minute intervals', () => {
    const domList = renderListContent({ interval: 15 })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })
  test('renders 20-minute intervals', () => {
    const domList = renderListContent({ interval: 20 })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })
  test('renders 30-minute intervals', () => {
    const domList = renderListContent({ interval: 30 })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })
  test('renders 60-minute intervals', () => {
    const domList = renderListContent({ interval: 60 })
    expect(extractTextFromDomList(domList)).toMatchSnapshot()
    fireEvent.click(document)
  })
})

describe('text input', () => {
  test('converts shorthand input to 24h formatted string on change', () => {
    const handleChange = jest.fn()
    const { getByPlaceholderText } = renderWithTheme(
      <InputTimeSelect onChange={handleChange} />
    )

    expect(handleChange).not.toHaveBeenCalled()

    const inputBox = getByPlaceholderText('Select time')
    fireEvent.click(inputBox)

    // Enter key input
    fireEvent.change(inputBox, { target: { value: '2pm' } })
    fireEvent.keyDown(inputBox, { key: 'Enter' })
    expect(handleChange).toHaveBeenLastCalledWith('14:00')

    // Tab key input
    fireEvent.change(inputBox, { target: { value: '3:3' } }) // incomplete time string
    fireEvent.keyDown(inputBox, { key: 'Tab' })
    expect(handleChange).toHaveBeenLastCalledWith('03:03')

    fireEvent.click(document)
  })
})

describe('keyboard nav ux', () => {
  test('highlights closest time to now when an unselected list is opened', () => {
    // NOTE: Date.now() mocked to 1580517818172 (Feb 01 2020 6:33 am)
    const domList = renderListContent({})
    const selected = domList.querySelector('[aria-selected="true"]')
    expect((selected as HTMLElement).textContent).toMatchSnapshot()
    fireEvent.click(document)
  })

  test('highlights selected value when list is opened', () => {
    const domList = renderListContent({
      onChange: jest.fn(),
      value: '14:15',
    })
    const selected = domList.querySelector('[aria-selected="true"]')
    expect((selected as HTMLElement).textContent).toMatchSnapshot()
    fireEvent.click(document)
  })

  test('highlights closest time to selected value when list is opened but value does not match provided options', () => {
    const domList = renderListContent({ onChange: jest.fn(), value: '16:38' })
    const selected = domList.querySelector('[aria-selected="true"]')
    expect((selected as HTMLElement).textContent).toMatchSnapshot()
    fireEvent.click(document)
  })
})
