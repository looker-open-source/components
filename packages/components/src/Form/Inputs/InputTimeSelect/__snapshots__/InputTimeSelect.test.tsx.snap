// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keyboard nav ux highlights closest time to now when an unselected list is opened 1`] = `"06:30 am"`;

exports[`keyboard nav ux highlights closest time to selected value when list is opened but value does not match provided options 1`] = `"04:45 pm"`;

exports[`keyboard nav ux highlights selected value when list is opened 1`] = `"02:15 pm"`;

exports[`prop: format formats options in 12 hour time 1`] = `
Array [
  "12:00 am",
  "12:15 am",
  "12:30 am",
  "12:45 am",
  "01:00 am",
  "01:15 am",
  "01:30 am",
  "01:45 am",
  "02:00 am",
  "02:15 am",
  "02:30 am",
  "02:45 am",
  "03:00 am",
  "03:15 am",
  "03:30 am",
  "03:45 am",
  "04:00 am",
  "04:15 am",
  "04:30 am",
  "04:45 am",
  "05:00 am",
  "05:15 am",
  "05:30 am",
  "05:45 am",
  "06:00 am",
  "06:15 am",
  "06:30 am",
  "06:45 am",
  "07:00 am",
  "07:15 am",
  "07:30 am",
  "07:45 am",
  "08:00 am",
  "08:15 am",
  "08:30 am",
  "08:45 am",
  "09:00 am",
  "09:15 am",
  "09:30 am",
  "09:45 am",
  "10:00 am",
  "10:15 am",
  "10:30 am",
  "10:45 am",
  "11:00 am",
  "11:15 am",
  "11:30 am",
  "11:45 am",
  "12:00 pm",
  "12:15 pm",
  "12:30 pm",
  "12:45 pm",
  "01:00 pm",
  "01:15 pm",
  "01:30 pm",
  "01:45 pm",
  "02:00 pm",
  "02:15 pm",
  "02:30 pm",
  "02:45 pm",
  "03:00 pm",
  "03:15 pm",
  "03:30 pm",
  "03:45 pm",
  "04:00 pm",
  "04:15 pm",
  "04:30 pm",
  "04:45 pm",
  "05:00 pm",
  "05:15 pm",
  "05:30 pm",
  "05:45 pm",
  "06:00 pm",
  "06:15 pm",
  "06:30 pm",
  "06:45 pm",
  "07:00 pm",
  "07:15 pm",
  "07:30 pm",
  "07:45 pm",
  "08:00 pm",
  "08:15 pm",
  "08:30 pm",
  "08:45 pm",
  "09:00 pm",
  "09:15 pm",
  "09:30 pm",
  "09:45 pm",
  "10:00 pm",
  "10:15 pm",
  "10:30 pm",
  "10:45 pm",
  "11:00 pm",
  "11:15 pm",
  "11:30 pm",
  "11:45 pm",
]
`;

exports[`prop: format formats options in 24 hour time 1`] = `
Array [
  "00:00",
  "00:15",
  "00:30",
  "00:45",
  "01:00",
  "01:15",
  "01:30",
  "01:45",
  "02:00",
  "02:15",
  "02:30",
  "02:45",
  "03:00",
  "03:15",
  "03:30",
  "03:45",
  "04:00",
  "04:15",
  "04:30",
  "04:45",
  "05:00",
  "05:15",
  "05:30",
  "05:45",
  "06:00",
  "06:15",
  "06:30",
  "06:45",
  "07:00",
  "07:15",
  "07:30",
  "07:45",
  "08:00",
  "08:15",
  "08:30",
  "08:45",
  "09:00",
  "09:15",
  "09:30",
  "09:45",
  "10:00",
  "10:15",
  "10:30",
  "10:45",
  "11:00",
  "11:15",
  "11:30",
  "11:45",
  "12:00",
  "12:15",
  "12:30",
  "12:45",
  "13:00",
  "13:15",
  "13:30",
  "13:45",
  "14:00",
  "14:15",
  "14:30",
  "14:45",
  "15:00",
  "15:15",
  "15:30",
  "15:45",
  "16:00",
  "16:15",
  "16:30",
  "16:45",
  "17:00",
  "17:15",
  "17:30",
  "17:45",
  "18:00",
  "18:15",
  "18:30",
  "18:45",
  "19:00",
  "19:15",
  "19:30",
  "19:45",
  "20:00",
  "20:15",
  "20:30",
  "20:45",
  "21:00",
  "21:15",
  "21:30",
  "21:45",
  "22:00",
  "22:15",
  "22:30",
  "22:45",
  "23:00",
  "23:15",
  "23:30",
  "23:45",
]
`;

exports[`prop: interval renders 5-minute intervals 1`] = `
Array [
  "12:00 am",
  "12:05 am",
  "12:10 am",
  "12:15 am",
  "12:20 am",
  "12:25 am",
  "12:30 am",
  "12:35 am",
  "12:40 am",
  "12:45 am",
  "12:50 am",
  "12:55 am",
  "01:00 am",
  "01:05 am",
  "01:10 am",
  "01:15 am",
  "01:20 am",
  "01:25 am",
  "01:30 am",
  "01:35 am",
  "01:40 am",
  "01:45 am",
  "01:50 am",
  "01:55 am",
  "02:00 am",
  "02:05 am",
  "02:10 am",
  "02:15 am",
  "02:20 am",
  "02:25 am",
  "02:30 am",
  "02:35 am",
  "02:40 am",
  "02:45 am",
  "02:50 am",
  "02:55 am",
  "03:00 am",
  "03:05 am",
  "03:10 am",
  "03:15 am",
  "03:20 am",
  "03:25 am",
  "03:30 am",
  "03:35 am",
  "03:40 am",
  "03:45 am",
  "03:50 am",
  "03:55 am",
  "04:00 am",
  "04:05 am",
  "04:10 am",
  "04:15 am",
  "04:20 am",
  "04:25 am",
  "04:30 am",
  "04:35 am",
  "04:40 am",
  "04:45 am",
  "04:50 am",
  "04:55 am",
  "05:00 am",
  "05:05 am",
  "05:10 am",
  "05:15 am",
  "05:20 am",
  "05:25 am",
  "05:30 am",
  "05:35 am",
  "05:40 am",
  "05:45 am",
  "05:50 am",
  "05:55 am",
  "06:00 am",
  "06:05 am",
  "06:10 am",
  "06:15 am",
  "06:20 am",
  "06:25 am",
  "06:30 am",
  "06:35 am",
  "06:40 am",
  "06:45 am",
  "06:50 am",
  "06:55 am",
  "07:00 am",
  "07:05 am",
  "07:10 am",
  "07:15 am",
  "07:20 am",
  "07:25 am",
  "07:30 am",
  "07:35 am",
  "07:40 am",
  "07:45 am",
  "07:50 am",
  "07:55 am",
  "08:00 am",
  "08:05 am",
  "08:10 am",
  "08:15 am",
  "08:20 am",
  "08:25 am",
  "08:30 am",
  "08:35 am",
  "08:40 am",
  "08:45 am",
  "08:50 am",
  "08:55 am",
  "09:00 am",
  "09:05 am",
  "09:10 am",
  "09:15 am",
  "09:20 am",
  "09:25 am",
  "09:30 am",
  "09:35 am",
  "09:40 am",
  "09:45 am",
  "09:50 am",
  "09:55 am",
  "10:00 am",
  "10:05 am",
  "10:10 am",
  "10:15 am",
  "10:20 am",
  "10:25 am",
  "10:30 am",
  "10:35 am",
  "10:40 am",
  "10:45 am",
  "10:50 am",
  "10:55 am",
  "11:00 am",
  "11:05 am",
  "11:10 am",
  "11:15 am",
  "11:20 am",
  "11:25 am",
  "11:30 am",
  "11:35 am",
  "11:40 am",
  "11:45 am",
  "11:50 am",
  "11:55 am",
  "12:00 pm",
  "12:05 pm",
  "12:10 pm",
  "12:15 pm",
  "12:20 pm",
  "12:25 pm",
  "12:30 pm",
  "12:35 pm",
  "12:40 pm",
  "12:45 pm",
  "12:50 pm",
  "12:55 pm",
  "01:00 pm",
  "01:05 pm",
  "01:10 pm",
  "01:15 pm",
  "01:20 pm",
  "01:25 pm",
  "01:30 pm",
  "01:35 pm",
  "01:40 pm",
  "01:45 pm",
  "01:50 pm",
  "01:55 pm",
  "02:00 pm",
  "02:05 pm",
  "02:10 pm",
  "02:15 pm",
  "02:20 pm",
  "02:25 pm",
  "02:30 pm",
  "02:35 pm",
  "02:40 pm",
  "02:45 pm",
  "02:50 pm",
  "02:55 pm",
  "03:00 pm",
  "03:05 pm",
  "03:10 pm",
  "03:15 pm",
  "03:20 pm",
  "03:25 pm",
  "03:30 pm",
  "03:35 pm",
  "03:40 pm",
  "03:45 pm",
  "03:50 pm",
  "03:55 pm",
  "04:00 pm",
  "04:05 pm",
  "04:10 pm",
  "04:15 pm",
  "04:20 pm",
  "04:25 pm",
  "04:30 pm",
  "04:35 pm",
  "04:40 pm",
  "04:45 pm",
  "04:50 pm",
  "04:55 pm",
  "05:00 pm",
  "05:05 pm",
  "05:10 pm",
  "05:15 pm",
  "05:20 pm",
  "05:25 pm",
  "05:30 pm",
  "05:35 pm",
  "05:40 pm",
  "05:45 pm",
  "05:50 pm",
  "05:55 pm",
  "06:00 pm",
  "06:05 pm",
  "06:10 pm",
  "06:15 pm",
  "06:20 pm",
  "06:25 pm",
  "06:30 pm",
  "06:35 pm",
  "06:40 pm",
  "06:45 pm",
  "06:50 pm",
  "06:55 pm",
  "07:00 pm",
  "07:05 pm",
  "07:10 pm",
  "07:15 pm",
  "07:20 pm",
  "07:25 pm",
  "07:30 pm",
  "07:35 pm",
  "07:40 pm",
  "07:45 pm",
  "07:50 pm",
  "07:55 pm",
  "08:00 pm",
  "08:05 pm",
  "08:10 pm",
  "08:15 pm",
  "08:20 pm",
  "08:25 pm",
  "08:30 pm",
  "08:35 pm",
  "08:40 pm",
  "08:45 pm",
  "08:50 pm",
  "08:55 pm",
  "09:00 pm",
  "09:05 pm",
  "09:10 pm",
  "09:15 pm",
  "09:20 pm",
  "09:25 pm",
  "09:30 pm",
  "09:35 pm",
  "09:40 pm",
  "09:45 pm",
  "09:50 pm",
  "09:55 pm",
  "10:00 pm",
  "10:05 pm",
  "10:10 pm",
  "10:15 pm",
  "10:20 pm",
  "10:25 pm",
  "10:30 pm",
  "10:35 pm",
  "10:40 pm",
  "10:45 pm",
  "10:50 pm",
  "10:55 pm",
  "11:00 pm",
  "11:05 pm",
  "11:10 pm",
  "11:15 pm",
  "11:20 pm",
  "11:25 pm",
  "11:30 pm",
  "11:35 pm",
  "11:40 pm",
  "11:45 pm",
  "11:50 pm",
  "11:55 pm",
]
`;

exports[`prop: interval renders 10-minute intervals 1`] = `
Array [
  "12:00 am",
  "12:10 am",
  "12:20 am",
  "12:30 am",
  "12:40 am",
  "12:50 am",
  "01:00 am",
  "01:10 am",
  "01:20 am",
  "01:30 am",
  "01:40 am",
  "01:50 am",
  "02:00 am",
  "02:10 am",
  "02:20 am",
  "02:30 am",
  "02:40 am",
  "02:50 am",
  "03:00 am",
  "03:10 am",
  "03:20 am",
  "03:30 am",
  "03:40 am",
  "03:50 am",
  "04:00 am",
  "04:10 am",
  "04:20 am",
  "04:30 am",
  "04:40 am",
  "04:50 am",
  "05:00 am",
  "05:10 am",
  "05:20 am",
  "05:30 am",
  "05:40 am",
  "05:50 am",
  "06:00 am",
  "06:10 am",
  "06:20 am",
  "06:30 am",
  "06:40 am",
  "06:50 am",
  "07:00 am",
  "07:10 am",
  "07:20 am",
  "07:30 am",
  "07:40 am",
  "07:50 am",
  "08:00 am",
  "08:10 am",
  "08:20 am",
  "08:30 am",
  "08:40 am",
  "08:50 am",
  "09:00 am",
  "09:10 am",
  "09:20 am",
  "09:30 am",
  "09:40 am",
  "09:50 am",
  "10:00 am",
  "10:10 am",
  "10:20 am",
  "10:30 am",
  "10:40 am",
  "10:50 am",
  "11:00 am",
  "11:10 am",
  "11:20 am",
  "11:30 am",
  "11:40 am",
  "11:50 am",
  "12:00 pm",
  "12:10 pm",
  "12:20 pm",
  "12:30 pm",
  "12:40 pm",
  "12:50 pm",
  "01:00 pm",
  "01:10 pm",
  "01:20 pm",
  "01:30 pm",
  "01:40 pm",
  "01:50 pm",
  "02:00 pm",
  "02:10 pm",
  "02:20 pm",
  "02:30 pm",
  "02:40 pm",
  "02:50 pm",
  "03:00 pm",
  "03:10 pm",
  "03:20 pm",
  "03:30 pm",
  "03:40 pm",
  "03:50 pm",
  "04:00 pm",
  "04:10 pm",
  "04:20 pm",
  "04:30 pm",
  "04:40 pm",
  "04:50 pm",
  "05:00 pm",
  "05:10 pm",
  "05:20 pm",
  "05:30 pm",
  "05:40 pm",
  "05:50 pm",
  "06:00 pm",
  "06:10 pm",
  "06:20 pm",
  "06:30 pm",
  "06:40 pm",
  "06:50 pm",
  "07:00 pm",
  "07:10 pm",
  "07:20 pm",
  "07:30 pm",
  "07:40 pm",
  "07:50 pm",
  "08:00 pm",
  "08:10 pm",
  "08:20 pm",
  "08:30 pm",
  "08:40 pm",
  "08:50 pm",
  "09:00 pm",
  "09:10 pm",
  "09:20 pm",
  "09:30 pm",
  "09:40 pm",
  "09:50 pm",
  "10:00 pm",
  "10:10 pm",
  "10:20 pm",
  "10:30 pm",
  "10:40 pm",
  "10:50 pm",
  "11:00 pm",
  "11:10 pm",
  "11:20 pm",
  "11:30 pm",
  "11:40 pm",
  "11:50 pm",
]
`;

exports[`prop: interval renders 15-minute intervals 1`] = `
Array [
  "12:00 am",
  "12:15 am",
  "12:30 am",
  "12:45 am",
  "01:00 am",
  "01:15 am",
  "01:30 am",
  "01:45 am",
  "02:00 am",
  "02:15 am",
  "02:30 am",
  "02:45 am",
  "03:00 am",
  "03:15 am",
  "03:30 am",
  "03:45 am",
  "04:00 am",
  "04:15 am",
  "04:30 am",
  "04:45 am",
  "05:00 am",
  "05:15 am",
  "05:30 am",
  "05:45 am",
  "06:00 am",
  "06:15 am",
  "06:30 am",
  "06:45 am",
  "07:00 am",
  "07:15 am",
  "07:30 am",
  "07:45 am",
  "08:00 am",
  "08:15 am",
  "08:30 am",
  "08:45 am",
  "09:00 am",
  "09:15 am",
  "09:30 am",
  "09:45 am",
  "10:00 am",
  "10:15 am",
  "10:30 am",
  "10:45 am",
  "11:00 am",
  "11:15 am",
  "11:30 am",
  "11:45 am",
  "12:00 pm",
  "12:15 pm",
  "12:30 pm",
  "12:45 pm",
  "01:00 pm",
  "01:15 pm",
  "01:30 pm",
  "01:45 pm",
  "02:00 pm",
  "02:15 pm",
  "02:30 pm",
  "02:45 pm",
  "03:00 pm",
  "03:15 pm",
  "03:30 pm",
  "03:45 pm",
  "04:00 pm",
  "04:15 pm",
  "04:30 pm",
  "04:45 pm",
  "05:00 pm",
  "05:15 pm",
  "05:30 pm",
  "05:45 pm",
  "06:00 pm",
  "06:15 pm",
  "06:30 pm",
  "06:45 pm",
  "07:00 pm",
  "07:15 pm",
  "07:30 pm",
  "07:45 pm",
  "08:00 pm",
  "08:15 pm",
  "08:30 pm",
  "08:45 pm",
  "09:00 pm",
  "09:15 pm",
  "09:30 pm",
  "09:45 pm",
  "10:00 pm",
  "10:15 pm",
  "10:30 pm",
  "10:45 pm",
  "11:00 pm",
  "11:15 pm",
  "11:30 pm",
  "11:45 pm",
]
`;

exports[`prop: interval renders 30-minute intervals 1`] = `
Array [
  "12:00 am",
  "12:30 am",
  "01:00 am",
  "01:30 am",
  "02:00 am",
  "02:30 am",
  "03:00 am",
  "03:30 am",
  "04:00 am",
  "04:30 am",
  "05:00 am",
  "05:30 am",
  "06:00 am",
  "06:30 am",
  "07:00 am",
  "07:30 am",
  "08:00 am",
  "08:30 am",
  "09:00 am",
  "09:30 am",
  "10:00 am",
  "10:30 am",
  "11:00 am",
  "11:30 am",
  "12:00 pm",
  "12:30 pm",
  "01:00 pm",
  "01:30 pm",
  "02:00 pm",
  "02:30 pm",
  "03:00 pm",
  "03:30 pm",
  "04:00 pm",
  "04:30 pm",
  "05:00 pm",
  "05:30 pm",
  "06:00 pm",
  "06:30 pm",
  "07:00 pm",
  "07:30 pm",
  "08:00 pm",
  "08:30 pm",
  "09:00 pm",
  "09:30 pm",
  "10:00 pm",
  "10:30 pm",
  "11:00 pm",
  "11:30 pm",
]
`;
